packages/shared/src/hooks/useFeedLayout.ts

```diff
@@ -106,7 +105,7 @@ export const useFeedLayout = ({
   );
 
   const shouldUseListFeedLayoutOnMobileTablet =
-    !isNullOrUndefined(isLaptop) && !isLaptop && isFeedIncludedInListLayout;
```

Comment:
Curious to know why we evaluate `isNullOrUndefined` here. It is producing some unwanted behavior.

Result if kept:
![Screenshot 2024-06-13 at 1 52 19 PM](https://github.com/dailydotdev/apps/assets/13744167/46851292-f8c2-4b6a-99e6-a796dab56eb3)

Reply:
It's because when server side rendering we don't know if it's laptop or mobile or whatever. I remember this issue, I think there was a bigger issue before, can't remember in detail though, but I think we agreed that we're fine this way since this can only happen if user goes directly to the link which would be SSR, and in mobile usually would go to this page by clicking on the internal links/buttons which would be rendered client side and not have this issue. 

I can try to find the thread where we agreed to this.

Reply:
Took me a while but here it is https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1716278512251279

Reply:
After further investigation, it seems the core issue is that the window was not yet ready when the evaluation happened. This is why some checks on load are showing differently from what is expected.

We can see all the issues happening from each previews below.

On this PR, we removed the `isNullOrUndefined` check which product the evaluation to be inaccurate:
https://mi-379.preview.app.daily.dev/bookmarks - the page is supposed to be using some other layout, but since the `isLaptop` is part of the evaluation, it is displaying differently.

On production, it works as intended, but there are some layout shifts happening:
https://app.daily.dev/bookmarks

On this preview, we wait for the window.load event before rendering, no noticeable layout shift and the layout is correct from the get-go:
https://refactor-feed-layout-hook.preview.app.daily.dev/bookmarks

Reply:
Would it affect other implementations as well?

Reply:
Yeah, mobile shouldn't be an issue, but on web it happens consistently.

Reply:
I think I will have to bring back the `isNullOrUndefined` and find a workaround for the issue that I was having. However, the bigger issue still remains.

Reply:
What's the concrete issue u were having @lee?

> However, the bigger issue still remains.

:hidethepain.....

Reply:
tagged the wrong `@lee` @nensidosari 😆

Reply:
That the first evaluation can be wrong when checking !isLaptop through the codebase which is the root of most first load layout shift.

Reply:
Ah yup, known issue now, does it affect anything in particular for this experiment and changes u've done that u think is blocking?

Reply:
I am getting this when the `is null or undefined` is present.

![Screenshot 2024-06-13 at 5 56 14 PM](https://github.com/dailydotdev/apps/assets/13744167/1cbae97a-d67f-4a29-8b61-7279495efb96)

Reply:
Cool. You mean non-blocking for this PR but blocking for the whole experiment?

Reply:
No I doubt, I'm not actually able to replicate so far on preview, but if it is like the previous ones I mentioned, it only happens if u move to this page via a link, not within the app clicks